### PR TITLE
feat: most relevant templates

### PR DESCRIPTION
**Pull request by GeronimoJohn**
# Description

### <samp>🤖 Generated by Copilot at d9e6e72</samp>

This pull request adds a new feature to the journeys-admin app that allows filtering and grouping templates by tags or featured status. It introduces a new component `TemplateSections` that renders different sections of templates based on the journeys data. It also adds test files, story files, and localization files for the new component and its subcomponents.

[- Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/34356579/todos/6580701359)

Depends on #1896 

# How should this PR be QA Tested?

- [ ] it should render the most relevant templates when the user pass in tagIds
- [ ] it should render the placeholder component when no templates are found

# Walkthrough

### <samp>🤖 Generated by Copilot at d9e6e72</samp>

*  Add `TemplateSections` component to render different sections of templates based on tags or featured status ([link](https://github.com/JesusFilm/core/pull/1916/files?diff=unified&w=0#diff-b3ab650ee0dc0ee335e53cf2f72cf2700dbc0890a665f9b7a7b608c6ce920650R5-R6), [link](https://github.com/JesusFilm/core/pull/1916/files?diff=unified&w=0#diff-b3ab650ee0dc0ee335e53cf2f72cf2700dbc0890a665f9b7a7b608c6ce920650R16), [link](https://github.com/JesusFilm/core/pull/1916/files?diff=unified&w=0#diff-3ecccae37a65f67176d50a97b5f5b69c90534d1b3ec82580ad08deb97aef6ce6R1), [link](https://github.com/JesusFilm/core/pull/1916/files?diff=unified&w=0#diff-8528e7b4c06335c3a765fccde78d68fff718a8b69d7384bea6782deeb45efb8aR1-R21))
*  Add `FeaturedAndNewTemplates` component to render templates that are either featured or recently published ([link](https://github.com/JesusFilm/core/pull/1916/files?diff=unified&w=0#diff-1880f963c96e52cb5647619202995dc83146b0ccaa82af8a3d97245ea204c784R1-R28), [link](https://github.com/JesusFilm/core/pull/1916/files?diff=unified&w=0#diff-8748926db339aed474e1c868f3b794f0a355d306879f45f26044763c4176bcceR1))
*  Add `MostRelevantTemplates` component to render templates that match the tags provided by the parent component ([link](https://github.com/JesusFilm/core/pull/1916/files?diff=unified&w=0#diff-39e756141f0162c975ad6a9471bd45f0d6d8cfb314f1230c31e30f59a7cf93acR1-R21), [link](https://github.com/JesusFilm/core/pull/1916/files?diff=unified&w=0#diff-193108d8d69c8f2c49c9ab5bf2559b9a6d59929b28764f3a7394d0c0f80c5be1R1))
*  Add `TemplateSection` component to render a section of journeys with a category heading and a horizontal scrollable layout ([link](https://github.com/JesusFilm/core/pull/1916/files?diff=unified&w=0#diff-0016fcb7bad744aecc9ef2abfad388fd2d490ec024dc331c2800e4f4bd2069b0R1-R29), [link](https://github.com/JesusFilm/core/pull/1916/files?diff=unified&w=0#diff-ab54d196a71a39df94e4e038d9e744bc4e9eefaf354fb8332897f5460d3c9557R1))
*  Add test files for each component using the `MockedProvider` and `render` functions to mock the GraphQL queries and assert the component output ([link](https://github.com/JesusFilm/core/pull/1916/files?diff=unified&w=0#diff-34dede514cb163916c72b5559b45fa638bdfd3ec7c44a5feb8a01c60a87bf15dR1-R119), [link](https://github.com/JesusFilm/core/pull/1916/files?diff=unified&w=0#diff-d71da8abb5aac78a7509caddb93e82ed7d6150fe8844d285cdeb7ee7433337c7R1-R100), [link](https://github.com/JesusFilm/core/pull/1916/files?diff=unified&w=0#diff-098622eda11a825df71054ce50442a8fc95638e4e36693b5e833b3d61f0153afR1-R61), [link](https://github.com/JesusFilm/core/pull/1916/files?diff=unified&w=0#diff-5ba9a6b26c37d6b4ed740618400d541896323f0c68da3bf932e64fe73d2e8fd9R1-R27))
*  Add story file for the `TemplateSections` component using the `MockedProvider` and mock journeys to showcase the component with different props ([link](https://github.com/JesusFilm/core/pull/1916/files?diff=unified&w=0#diff-a8df0dbccc05494a8cf4897c19491e20d422103f214e396b9af94a836422cb6dR1-R120))
*  Add localized strings for the category names of the template sections to the English locale file ([link](https://github.com/JesusFilm/core/pull/1916/files?diff=unified&w=0#diff-acd82a7e9a7b8b41e5d2ba5351ff70551bf8cb53d9a2dd9d010acedae21d2a2dR234-R235))

### CODE REVIEW

File: apps/journeys-admin/src/components/TemplateSections/TemplateSections.tsx [hunk 1]
```diff
@@ -2,11 +2,20 @@ import Stack from '@mui/material/Stack'
 import { ReactElement } from 'react'
 
 import { FeaturedAndNewTemplates } from './FeaturedAndNewTemplates'
+import { MostRelevantTemplates } from './MostRelevantTemplates'
 
-export function TemplateSections(): ReactElement {
+interface Props {
+  tags?: string[]
+}
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/TemplateSections/TemplateSections.tsx:9]: replace with check the url params

File: apps/journeys-admin/src/components/TemplateSections/MostRelevantTemplates/MostRelevantTemplates.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { ReactElement } from 'react'
+import { useTranslation } from 'react-i18next'
+
+import { useJourneysQuery } from '../../../libs/useJourneysQuery'
+import { TemplateSection } from '../TemplateSection'
+
+interface Props {
+  tags?: string[]
```
Comment (storyworks) [inline apps/journeys-admin/src/components/TemplateSections/MostRelevantTemplates/MostRelevantTemplates.tsx:8]: Could just get this from useRouter hook?

File: apps/journeys-admin/src/components/TemplateSections/MostRelevantTemplates/MostRelevantTemplates.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { ReactElement } from 'react'
+import { useTranslation } from 'react-i18next'
+
+import { useJourneysQuery } from '../../../libs/useJourneysQuery'
+import { TemplateSection } from '../TemplateSection'
+
+interface Props {
```
Comment (storyworks) [inline apps/journeys-admin/src/components/TemplateSections/MostRelevantTemplates/MostRelevantTemplates.tsx:7]: Naming convention is `[NameOfComponent]Props`

File: apps/journeys-admin/src/components/TemplateSections/MostRelevantTemplates/MostRelevantTemplates.tsx (hunk 2 above)
Comment (GeronimoJohn) [reply apps/journeys-admin/src/components/TemplateSections/MostRelevantTemplates/MostRelevantTemplates.tsx:8]: Yep was thinking that, build this before wihtout it

File: apps/journeys-admin/src/components/TemplateSections/MostRelevantTemplates/MostRelevantTemplates.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    )
+    expect(getByRole('heading', { name: 'Most Relevant' })).toBeInTheDocument()
+    await waitFor(() => {
+      const cards = getAllByTestId(/journey-/)
+      expect(cards[0]).toHaveTextContent('Template 1')
+      expect(cards[1]).toHaveTextContent('Template 2')
+    })
+  })
+
+  it('should not render Most Relevant Templates when TagIds are not present', async () => {
```
Comment (storyworks) [inline apps/journeys-admin/src/components/TemplateSections/MostRelevantTemplates/MostRelevantTemplates.spec.tsx:111]: Rename to "should not render templates when no tag ids" ? I read this to mean - don't render the entire component when there are no tag ids. But we still render the heading which is correctly shown in your test (so I could eventually find out).

Rename the first test to match? "should render templates from tag id".

Actually, I've just realised the mocking of this test is incorrect. You're fetching tagIds: 'tagId1', but the returned data have tag id's that are in uuid format. Fix this here and in the other component tests you've already merged to prod

File: apps/journeys-admin/src/components/TemplateSections/MostRelevantTemplates/MostRelevantTemplates.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+      query: {
+        tags: [
+          '767561ca-3f63-46f4-b2a0-3a37f891632a',
+          '4e6cf9db-0928-4e34-8e63-8f62bddf87d4'
+        ]
+      }
+    } as unknown as NextRouter)
+
+    const { getByRole, getAllByTestId } = render(
+      <MockedProvider
```
Comment (Kneesal) [inline apps/journeys-admin/src/components/TemplateSections/MostRelevantTemplates/MostRelevantTemplates.spec.tsx:77]: `
  \  result = jest.fn(() => ({
      data: {
        journeys: [
          template,
          {
            ...template,
            id: 2,
            title: 'Template 2'
          }
        ]
      }
    }))
`

File: apps/journeys-admin/src/components/TemplateSections/MostRelevantTemplates/MostRelevantTemplates.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+        mocks={[
+          {
```
Comment (Kneesal) [inline apps/journeys-admin/src/components/TemplateSections/MostRelevantTemplates/MostRelevantTemplates.spec.tsx:79]:
```
`
  \  result = jest.fn(() => ({
      data: {
        journeys: [
          template,
          {
            ...template,
            id: 2,
            title: 'Template 2'
          }
        ]
      }
    }))
`
```

File: apps/journeys-admin/src/components/TemplateSections/MostRelevantTemplates/MostRelevantTemplates.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+            request: {
+              query: GET_JOURNEYS,
```
Comment (Kneesal) [inline apps/journeys-admin/src/components/TemplateSections/MostRelevantTemplates/MostRelevantTemplates.spec.tsx:81]: use result const

File: apps/journeys-admin/src/components/TemplateSections/MostRelevantTemplates/MostRelevantTemplates.spec.tsx [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+      >
+        <MostRelevantTemplates />
+      </MockedProvider>
+    )
+    expect(getByRole('heading', { name: 'Most Relevant' })).toBeInTheDocument()
+    await waitFor(() => {
+      const cards = getAllByTestId(/journey-/)
+      expect(cards[0]).toHaveTextContent('Template 1')
+      expect(cards[1]).toHaveTextContent('Template 2')
+    })
```
Comment (Kneesal) [inline apps/journeys-admin/src/components/TemplateSections/MostRelevantTemplates/MostRelevantTemplates.spec.tsx:116]: expect(result).toHaveBeenCalled()

File: apps/journeys-admin/src/components/TemplateSections/MostRelevantTemplates/MostRelevantTemplates.spec.tsx [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+                  template: true,
+                  orderByRecent: true,
+                  tagIds: [
+                    '767561ca-3f63-46f4-b2a0-3a37f891632a',
+                    '4e6cf9db-0928-4e34-8e63-8f62bddf87d4'
+                  ]
+                }
+              }
+            },
+            result: {
```
Comment (Kneesal) [inline apps/journeys-admin/src/components/TemplateSections/MostRelevantTemplates/MostRelevantTemplates.spec.tsx:93]: use result const

File: apps/journeys-admin/src/components/TemplateSections/MostRelevantTemplates/MostRelevantTemplates.spec.tsx [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+  it('should render templates from tag id', async () => {
+    mockUseRouter.mockReturnValue({
+      query: {
+        tags: [
+          '767561ca-3f63-46f4-b2a0-3a37f891632a',
+          '4e6cf9db-0928-4e34-8e63-8f62bddf87d4'
+        ]
+      }
+    } as unknown as NextRouter)
+
```
Comment (Kneesal) [inline apps/journeys-admin/src/components/TemplateSections/MostRelevantTemplates/MostRelevantTemplates.spec.tsx:75]:
```  tsx
   const  result = jest.fn(() => ({
      data: {
        journeys: [
          template,
          {
            ...template,
            id: 2,
            title: 'Template 2'
          }
        ]
      }
    }))

**Review by Kneesal — COMMENTED** [review]
I like this component

**Review by Kneesal — DISMISSED** [review]

**Review by Kneesal — APPROVED** [review]

File: apps/journeys-admin/src/components/TemplateSections/MostRelevantTemplates/MostRelevantTemplates.tsx [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+import { useJourneysQuery } from '../../../libs/useJourneysQuery'
+import { TemplateSection } from '../TemplateSection'
+
+export function MostRelevantTemplates(): ReactElement {
+  const { t } = useTranslation('apps-journeys-admin')
+  const router = useRouter()
+  const tags = router.query.tags as string[]
+
+  const { data } = useJourneysQuery({
+    where: { template: true, orderByRecent: true, tagIds: tags }
```
Comment (storyworks) [inline apps/journeys-admin/src/components/TemplateSections/MostRelevantTemplates/MostRelevantTemplates.tsx:14]: In future PRs, update this to get tags from parent component

**Review by storyworks — APPROVED** [review]

File: apps/journeys-admin/src/components/TemplateSections/MostRelevantTemplates/MostRelevantTemplates.tsx [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+          journeys={data?.journeys}
+        />
+      ) : (
+        <Stack
+          gap={3}
+          sx={{
+            borderRadius: 4,
+            border: 1,
+            width: '100%',
+            height: 132,
```
Comment (storyworks) [inline apps/journeys-admin/src/components/TemplateSections/MostRelevantTemplates/MostRelevantTemplates.tsx:34]: Why hardcode the height? Is padding + having content not enough?

Shouldn't need "display: flex" if you're using stack.

File: apps/journeys-admin/src/components/TemplateSections/MostRelevantTemplates/MostRelevantTemplates.tsx [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+            padding: 8,
+            display: 'flex',
+            justifyContent: 'center'
+          }}
+        >
+          <Typography variant="h6">
+            {t('No template that fully matches your search criteria.')}
```
Comment (storyworks) [inline apps/journeys-admin/src/components/TemplateSections/MostRelevantTemplates/MostRelevantTemplates.tsx:41]: "No template fully matches your search criteria."

Correct the english from the design

File: apps/journeys-admin/src/components/TemplateSections/MostRelevantTemplates/MostRelevantTemplates.tsx [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+
+  return (
+    <>
+      {data != null && data?.journeys?.length > 0 ? (
+        <TemplateSection
+          category={t('Most Relevant')}
+          journeys={data?.journeys}
+        />
+      ) : (
+        <Stack
```
Comment (storyworks) [inline apps/journeys-admin/src/components/TemplateSections/MostRelevantTemplates/MostRelevantTemplates.tsx:28]: Use paper when building a container with background / outline. Stack/Grid is used for layout only.

https://mui.com/material-ui/react-paper/#variants

Paper with elevation 0 and outline could work. Might need to still edit border radius.

**Review by storyworks — APPROVED** [review]

**Pull request merged**